### PR TITLE
Larry local branch

### PR DESCRIPTION
**Pull request by larrykimutai**

### CODE REVIEW

**Review by DAndreBrown — APPROVED** [review]

File: pii_data.py [hunk 1]
```diff
@@ -58,16 +58,29 @@ def has_ipv4(self, anonymize=False):
...
-                          r'(\b[0-9a-fA-F]{0,4}\b)?:(\b[0-9a-fA-F]{0,4}\b)?:'
-                          r'(\b[0-9a-fA-F]{0,4}\b)?:(\b[0-9a-fA-F]{0,4}\b)?$)', self)
+    def has_ipv6(self, anonymize=False):
+
+        ipv6, count = re.subn(r'((\w((?:[0-9a-fA-F]?){0,4}:)|(:)){7}((?:[0-9a-fA-F]?){0,4}))([^:[0-9a-fA-F]])*',
+                              '[iPv6 address]', self)
+
+        ipv6, count0 = re.subn(r'((\w((?:[0-9a-fA-F]?){0,4}:)|(:)){7}((?:[0-9a-fA-F]?){0,4}))([^:[0-9a-fA-F]])*',
+                               '[iPv6 address]', ipv6)
         if self.__eq__('0:0:0:0:0:0:0:0') | self.__eq__(':::::::'):
```
Comment (claesmk) [inline pii_data.py:68]: You've used a [bitwise](https://www.w3schools.com/python/gloss_python_bitwise_operators.asp) operator here - you don't want to do that.  Instead you should use or like this:
```
        if self.__eq__('0:0:0:0:0:0:0:0') or self.__eq__(':::::::'):
```

File: test_pii_data.py [hunk 2]
```diff
@@ -176,28 +176,32 @@ def test_has_ipv4(self):
     def test_has_ipv6(self):
         test_data = Pii('2001:0db8:85a3:0000:0000:8a2e:0370:7334')
         self.assertTrue(test_data.has_ipv6())  # test a valid address
+        print("\n" + test_data.has_ipv6(True))
```
Comment (claesmk) [inline test_pii_data.py:179]: This print does not actually test the anonymize functionality.  You want to do something like this:
```
        print("\n" + test_data.has_ipv6(True))
        self.assertEqual('[iPv6 address]',
                         test_data.has_ipv6(True))
```

**Review by claesmk — CHANGES_REQUESTED** [review]
See embedded comments

**Review by claesmk — APPROVED** [review]
Looks good

**Pull request merged**